application/Profile/LocalProfile.py

```
@@ -37,42 +38,57 @@
-        raise NotImplementedError(f'Method {__name__} should not be invoke from {__class__}')
+        not_implemented_error = f'Method {__name__} should not be invoke from {__class__}'
+        logger.error(not_implemented_error)
+        raise NotImplementedError(not_implemented_error)
 
     def send_activation_email(self, username):
-        raise NotImplementedError(f'Method {__name__} should not be invoke from {__class__}')
+        not_implemented_error = f'Method {__name__} should not be invoke from {__class__}'
+        logger.error(not_implemented_error)
+        raise NotImplementedError(not_implemented_error)
 
     def __init__(self):
```

Ruff

Missing return type annotation for special method `__init__`

Add return type annotation: `None`

(ANN204)